src/actions/preferences.js

```diff
@@ -0,0 +1,25 @@
+import { fetchPreferences } from '../utils/preferences-api';
+
+// Action Types
+export const PREFERENCES_LOADED = 'PREFERENCES_LOADED';
+export const TOGGLE_SHOW_DATASET_PREVIEWS = 'TOGGLE_SHOW_DATASET_PREVIEWS';
+
+// Action Creators
+export const preferencesLoaded = (preferences) => ({
+  type: PREFERENCES_LOADED,
+  payload: preferences,
+});
+
+export const toggleShowDatasetPreviews = (showDatasetPreviews) => ({
+  type: TOGGLE_SHOW_DATASET_PREVIEWS,
+  showDatasetPreviews,
+});
```

Comment:
@SajidAlamQB 
Can we combine `preferencesLoaded` and `toggleShowDatasetPreviews` to have single action like `updateDatasetPreviews`/ `UPDATE_DATASETS_PREVIEWS` as both are updating `showDatasetPreviews` in Redux state.